packages/form-core/src/form-group/FormGroupMixin.js

```diff
@@ -317,21 +317,39 @@ const FormGroupMixinImplementation = superclass =>
       });
     }
 
+    /**
+     * A filter function which will exclude a form field when returning false
+     * By default, exclude form fields which are disabled
+     *
+     * The type is be passed as well for more fine grained control, e.g.
+     * distinguish the filter when fetching modelValue versus serializedValue
+     *
+     * @param {FormControl} el
+     * @param {string} type
+     * @returns {boolean}
+     */
+    // eslint-disable-next-line class-methods-use-this, no-unused-vars
+    _getFromAllFormElementsFilter(el, type) {
+      return !el.disabled;
+    }
+
     /**
      * Gets a keyed be name object for requested property (like modelValue/serializedValue)
      * @param {string} property
      * @returns {{[name:string]: any}}
      */
-    _getFromAllFormElements(property, filterFn = (/** @type {FormControl} */ el) => !el.disabled) {
+    _getFromAllFormElements(property) {
       const result = {};
       // @ts-ignore [allow-protected]: allow Form internals to access this protected method
       this.formElements._keys().forEach(name => {
         const elem = this.formElements[name];
         if (elem instanceof FormControlsCollection) {
-          result[name] = elem.filter(el => filterFn(el)).map(el => el[property]);
-        } else if (filterFn(elem)) {
+          result[name] = elem
+            .filter(el => this._getFromAllFormElementsFilter(el, property))
+            .map(el => el[property]);
+        } else if (this._getFromAllFormElementsFilter(elem, property)) {
           if (typeof elem._getFromAllFormElements === 'function') {
-            result[name] = elem._getFromAllFormElements(property, filterFn);
+            result[name] = elem._getFromAllFormElements(property);
```

Comment:
This kind of assumes that the filter is always there for one single purpose (serialization). We would loose the ability to filtern (on form/fieldset level) to filter for readonly for instance.

We will look for a way to keep that option, while still providing a default serialize condition for every FormControl.

Maybe it's also an idea to rename `_getFromAllFormElementsFilter()` to `_serializeCondition()` ?

Reply:
I disagree. You are passing the property to the filter function, and inside the filter function you can do different things for different properties. So it could be serializeCondition, it could be modelValueCondition, initialModelValueCondition, etc. etc. and in the filter function you can specify what you want to filter for on the element, e.g. disabled, readonly, checked, etc.

The test should make that a bit clearer

Reply:
I think the problem is that on fieldset or form level you, generecially speaking, don't know anyhting about your children FormControls (they could be added dynamically by 3drd parties that consume your form). You should still be able to get all readonly (or non readonly or any other criterium) elements of all your children and their children etc. without being dependent on implementation details (the condition in `_getFromAllFormElementsFilter` in this case) you don't know about.

Reply:
I don't understand what this PR changes about any of that. The only thing it does is pull out the filter condition function into a protected method which makes it easier to override without having to override the entire `_getFromAllFormElements` method.

Reply:
I mean that if the form below (that wouldn't know anything about the implementation of `_getFromAllFormElementsFilter` calls `_getFromAllFormElements('modelValue')`, it gets a kind of unexpected result.
I think the default condition could highly depend on the first argument (modelValue in example above, but could be any prop). If a FormControl wants to totally be independent of its parent fieldset/form, it should probably not provide a name.

Also, I want to be able to do this in my form: `_getFromAllFormElements('modelValue', (el) => !el.getAttribute('field-type') === 'helper')`


(since the public attr field-type is one I provided myself.)

```html
<lion-form id="myForm">

  <my-field></my-field>
  <!-- has . _getFromAllFormElementsFilter="${(el) => !el.disabled}" -->
  
  <my-other-field field-type="helper""></my-other-field>
    <!-- has . _getFromAllFormElementsFilter="${(el) => !el.readonly}" -->


</lion-form>
```

Reply:
> Also, I want to be able to do this in my form: `_getFromAllFormElements('modelValue', (el) => !el.getAttribute('field-type') === 'helper')`

Yeah that makes sense, we can keep the 2nd argument filter condition function that takes precedence over the protected filter method? Would that suffice?

So you can do

```js
_getFromAllFormElements('modelValue'); // uses this._getFromAllFormElementsFilter();
_getFromAllFormElements('modelValue', (el) => !el.getAttribute('field-type') === 'helper'); // uses the 2nd arg as the filter function
```

Reply:
Pushed the change in a separate commit @tlouisse . Let me know if it's alright, then I can rebase / squash commits